### ISSUE TITLE
Ch3 Notebook: Trend Follow Strategy

### ISSUE THREAD

**Issue by Jackal08** [BlackArbsCEO/Adv_Fin_ML_Exercises#24]
I just noticed now that your input vector for the random forest (trend following) has only one feature. ```X = ma_side.values.reshape(-1,1)```

This will be a problem since the model wont be able to correctly split on nodes (there is only one feature). So at best the tree can do a single split, 1 or 0. The model doesn't have enough information to learn from the features. Maybe adding some of the previous sides as well as things like auto correlation would help. 

I suspect that its the reason why the trend following RF always predicts to always trade.

Would love to hear your thoughts.
Best Regards
Jacques Joubert

**Comment by BlackArbsCEO** [comment]
Yes, I think you're right. In this notebook I was just doing the basic implementations. Later on in the bollinger bands strategy additional features are included. Feel free to submit a pull request with additional features included. You can create a new notebook with just the meta-labeling and feature engineering if you like.

**Comment by Jackal08** [comment]
I see thank you. I found that one of the problems we were having was that our sample was too small and the positive trades were an imbalanced class. It made a big difference when we corrected it. 

The CUSUM filters used to filter the events also gives us a very sparse set, especially if we set the min_ret to say 0.01. Still trying to work on that part. We are still learning as we go, thanks for this great source.
